src/snowflake/snowpark/_internal/utils.py

```diff
@@ -42,6 +42,8 @@
 SNOWFLAKE_ID_PATTERN = (
     f"({SNOWFLAKE_UNQUOTED_ID_PATTERN}|{SNOWFLAKE_QUOTED_ID_PATTERN})"
 )
+SNOWFLAKE_CASE_INSENSITIVE_QUOTED_ID_PATTERN = r'("([A-Z_][A-Z0-9_\$]{0,255})")'
+SNOWFLAKE_CASE_INSENSITIVE_UNQUOTED_SUFFIX_PATTERN = r"([a-zA-Z0-9_\$]{0,255})"
```

Comment:
Just curious, why does unquoted suffix include lower case but not quoted id?

Reply:
It has no quote. The different from an ID is an insensitive suffix can start with a digit (0-9) while an ID can't.